Documentation/workflow/DevelopmentTips.md

```diff
@@ -1,37 +1,292 @@
+
+### Option B: Upload the library with symbols to the update directory
+
+This option is useful for testing a number of different `libmonosgen-2.0.so`
+versions quickly without rebuilding or redeploying the app, but it requires a
+little care to complete the `adb` steps correctly on the command line.
+
+ 1. Push the appropriate architecture of `libmonosgen-2.0.d.so` into the
+    application's update directory and rename it to `libmonosgen-2.0.so`:
+
+        $ adb push libmonosgen-2.0.d.so \
+            /data/local/tmp/libmonosgen-2.0.so && \
+          adb shell run-as Mono.Android_Tests cp /data/local/tmp/libmonosgen-2.0.so \
+            /data/data/Mono.Android_Tests/files/.__override__/
+
+ 2. Ensure all users have execute permissions on the application's data
+    directory:
+
+        $ adb shell run-as Mono.Android_Tests \
+            chmod a+x /data/data/Mono.Android_Tests/
+
+    This will allow LLDB to re-download `libmonosgen-2.0.so` and load the
+    symbols from it.
+
+ 3. Run the app and attach LLDB.
+
+### Option C: Manually load the library with symbols
+
+This option allows testing an existing debuggable APK without pushing anything
+new to the device.  The other options are usually more convenient, but loading
+the symbols by hand might be useful in some cases.
+
+ 1. After attaching LLDB to the app, add the appropriate architecture of
+    `libmonosgen-2.0.d.so` into LLDB with a command like:
+
+        (lldb) image add ~/Downloads/lib/xamarin.android/xbuild/Xamarin/Android/lib/arm64-v8a/libmonosgen-2.0.d.so
+
+ 2. Find the current in-memory address of the `.text` section of
+    `libmonosgen-2.0`.  For example, for a 64-bit app that's using the shared
+    runtime, run the following command:
+
+        (lldb) image dump sections libmonosgen-64bit-2.0.so
+
+    Look for the row of the table that shows "code" as the "Type":
+
+        SectID     Type             Load Address                             Perm File Off.  File Size  Flags      Section Name
+        ---------- ---------------- ---------------------------------------  ---- ---------- ---------- ---------- ----------------------------
+        0x0000000a code             [0x00000071106c4e80-0x0000007110932674)  r-x  0x0002ee80 0x0026d7f4 0x00000006 libmonosgen-64bit-2.0.so..text
+
+ 3. Load the `.text` section from `libmonosgen-2.0.d.so` at the in-memory
+    starting memory address of the `.text` section:
+
+        (lldb) image load -f libmonosgen-2.0.d.so .text 0x00000071106c4e80
+
+## Attaching LLDB or GDB without Android Studio
+
+### Attaching LLDB using mono/lldb-binaries on macOS
+
+Download the precompiled `lldb` and `lldb-server` binaries from
+<https://github.com/mono/lldb-binaries/releases>, and follow the instructions
+within [README.md][lldb-readme].
+
+[lldb-readme]: https://github.com/mono/lldb-binaries/blob/master/README.md
+
+### Attaching GDB using Visual Studio on Windows
```

Comment:
This section should be promoted to a "top-level" `#` section, and should come *second*, before the Android Studio section.